### PR TITLE
Add instructions on dynamic linking

### PR DESCRIPTION
**Pull request by jonathanmetzman**
I think @kcc recommended against this because it will probably encourage more people to use dynamic linking.

### CODE REVIEW

File: docs/further-reading/fuzzer_environment.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ to `'$ORIGIN'`, the path of the binary (see the discussion of `'$ORIGIN'` [here]
 A fuzzer can be instructed to load libraries relative to `'$ORIGIN'` during compilation (e.g. `-Wl,-rpath,'$ORIGIN/lib'` )
 or afterwards using `chrpath` (e.g. `chrpath -r '$ORIGIN/lib' $OUT/$fuzzerName`). Note that `'$ORIGIN'` should be surronded
 by single quotes because it is not an environment variable like `$OUT` that can be retreived during execution of `build.sh`.
-It's value is retrieved during execution of the binary.
+Its value is retrieved during execution of the binary.
```
Comment (oliverchang) [inline docs/further-reading/fuzzer_environment.md]: Maybe add a note that they can verify RPATHs are valid by using ldd?

File: docs/further-reading/fuzzer_environment.md [hunk 2]
```diff
@@ -39,6 +39,14 @@ All build artifacts needed during fuzz target execution should be inside the
 Everything else is ignored (e.g. artifacts in `$WORK`, `$SRC`, etc) and hence
 is not available in the execution environment.
 
+We strongly recommend static linking because it just works. 
+However dynamic linking can work if shared objects are included in the `$OUT` directory and are loaded relative
+to `'$ORIGIN'`, the path of the binary (see the discussion of `'$ORIGIN'` [here](http://man7.org/linux/man-pages/man8/ld.so.8.html)).
+A fuzzer can be instructed to load libraries relative to `'$ORIGIN'` during compilation (e.g. `-Wl,-rpath,'$ORIGIN/lib'` )
+or afterwards using `chrpath` (e.g. `chrpath -r '$ORIGIN/lib' $OUT/$fuzzerName`). Note that `'$ORIGIN'` should be surronded
+by single quotes because it is not an environment variable like `$OUT` that can be retreived during execution of `build.sh`.
```
Comment (oliverchang) [inline docs/further-reading/fuzzer_environment.md]: nit: fix spelling of "retreived"

File: docs/further-reading/fuzzer_environment.md (hunk 1 above)
Comment (inferno-chromium) [reply docs/further-reading/fuzzer_environment.md]: +1

File: docs/further-reading/fuzzer_environment.md (hunk 2 above)
Comment (inferno-chromium) [reply docs/further-reading/fuzzer_environment.md]: +1

File: docs/further-reading/fuzzer_environment.md [hunk 3]
```diff
@@ -39,6 +39,14 @@ All build artifacts needed during fuzz target execution should be inside the
...
+Its value is retrieved during execution of the binary.
+
```
Comment (inferno-chromium) [inline docs/further-reading/fuzzer_environment.md]: Can you add a link to example build.sh (like facebook one).

Comment (jonathanmetzman) [reply docs/further-reading/fuzzer_environment.md]: Facebook one is  a weird case because patchelf is called by `find`. Will do php which uses `chrpath`.

**Comment by Dor1s** [comment]
> it will probably encourage more people to use dynamic linking.

+1, not sure how much value vs. potential headache we will get with this, your call.

**Pull request merged**